core/src/main/java/org/pentaho/di/core/logging/LoggingRegistry.java

```diff
@@ -157,25 +193,88 @@ public LoggingObjectInterface findExistingLoggingSource( LoggingObjectInterface
     return found;
   }
 
+  /**
+   * Determine if the "found" existing LoggingSource matches the loggingSource that is be attempted to register.
+   * @param loggingSource  LoggingObjectInterface that is being registered
+   * @param foundSource  LoggingObjectInterface which was "found"
+   * @return LogChannelID if found otherwise empty string.
+   */
+  private String determineExistingLoggingSource( LoggingObjectInterface loggingSource, LoggingObjectInterface foundSource ) {
+    LoggingObjectInterface foundParent = foundSource.getParent();
+    LoggingObjectInterface loggingSourceParent = loggingSource.getParent();
+    String foundLogChannelId = foundSource.getLogChannelId();
+
+    if ( !foundLogChannelId.isEmpty() ) {
+
+      if ( foundParent != null && loggingSourceParent != null ) {
+        String foundParentLogChannelId = foundParent.getLogChannelId();
+        String sourceParentLogChannelId = loggingSourceParent.getLogChannelId();
+
+        if ( foundParentLogChannelId != null && foundParentLogChannelId.equals( sourceParentLogChannelId ) ) {
+          return foundLogChannelId;
+        }
+      }
+
+      if ( foundParent == null && loggingSourceParent == null  ) {
+        return foundLogChannelId;
+      }
+    }
+    return "";
+  }
+
+  /**
+   * Returns a LoggingObjectInterface based on a LogChannelID which is the key for the LoggingRegistry.
+   * @param logChannelId  the logChannelID to search for.
+   * @return  the LoggingObjectInterface if found, null otherwise.
+   */
   public LoggingObjectInterface getLoggingObject( String logChannelId ) {
     return this.map.get( logChannelId );
   }
 
+  @VisibleForTesting
+  public int getRegistryMapSize() {
+    return this.map.size();
+  }
+
+
+  /**
+   * @deprecated This is unsafe call and references to this method will be remove.
+   */
+  @Deprecated
   public Map<String, LoggingObjectInterface> getMap() {
     return this.map;
   }
 
+  @VisibleForTesting
+  public void setMaxSize( int maxSize ) {
```

Comment:
Could this method, and setPurgeTimeout, be package-private?  Still testable, and reduces risk of misuse.

Reply:
Fixed